pkg/kubelet/kubelet.go

```diff
@@ -1712,6 +1715,14 @@ func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	return nil
 }
 
+func (kl *Kubelet) setPodStatusIfNotTerminal(pod *v1.Pod, status v1.PodStatus) bool {
+	if (status.Phase == v1.PodSucceeded || status.Phase == v1.PodFailed) && kl.podWorkers.CouldHaveRunningContainers(pod.UID) {
```

Comment:
I think this would prevent pods with two containers from both reporting their exit code if one fails early.